pytorch_lightning/core/lightning.py

```diff
@@ -1724,26 +1736,37 @@ def module_arguments(self) -> dict:
 
 
 def _collect_init_args(frame, path_args: list) -> list:
-    """Recursive search for all children."""
-    if '__class__' in frame.f_locals:
-        local_args = dict(frame.f_locals)
-        local_args.update(local_args.get('kwargs', {}))
-        local_args = {k: v for k, v in local_args.items()
-                      if k not in ('args', 'kwargs', 'self', '__class__', 'frame', 'frame_args')}
-        # if 'hparams' in local_args:
-        #     # back compatible hparams as single argument
-        #     hparams = local_args.get('hparams')
-        #     local_args.update(vars(hparams) if isinstance(hparams, Namespace) else hparams)
+    """
+    Recursively collects the arguments passed to the child constructors in the inheritance tree.
+
+    Args:
+        frame: the current stack frame
+        path_args: a list of dictionaries containing the constructor args in all parent classes
+
+    Return:
+          A list of dictionaries where each dictionary contains the arguments passed to the
+          constructor at that level. The last entry corresponds to the constructor call of the
+          most specific class in the hierarchy.
+    """
+    _, _, _, local_vars = inspect.getargvalues(frame)
+    if '__class__' in local_vars:
+        cls = local_vars['__class__']
+        spec = inspect.getfullargspec(cls.__init__)
+        init_parameters = inspect.signature(cls.__init__).parameters
+        self_identifier = spec.args[0]        # "self" unless user renames it (always first arg)
+        varargs_identifier = spec.varargs     # by convention this is named "*args"
+        kwargs_identifier = spec.varkw        # by convention this is named "**kwargs"
```

Comment:
Not sure about this kwargs ignoring but tests are passing... As kwargs can be used as passing argument of parent from the latest child...

Reply:
see my answer to Justus, I think you are asking the same as he. correct me if I'm wrong, it was like this before.

Reply:
yeah, we wrote it in parallel :]